[SG-183] Add-drop widget

Jaesin opened the pull request:
For landing page sections, the button to add content to sections will use the new and improved widget. The initial add entry will appear as "Add" and subsequent entries will appear by their paragraph type name.

![add-section-content](https://user-images.githubusercontent.com/159693/41491667-788681e8-70af-11e8-8835-c995f2dcf7fc.gif)

josh-chou commented on web/modules/custom/sfgov_admin/src/Plugin/Field/FieldWidget/SfgovParagraphsWidget.php:
```diff
@@ -0,0 +1,101 @@
+    return $elements;
+  }
+
+}
```
Can you add tests to cover this functionality?  Thanks!

Jaesin replied:
Hi @josh-chou,

Sure. What type of tests would you like to see for this? 

I know behat tests are running but it doesn't look unit tests (such as `Drupal\Tests\sfgov_departments`) are running.

I would typically write unit/kernel/functional or functional js tests in Drupal's frameworks.

josh-chou replied:
Any type of test coverage would be great.  I was investigating behat to run the sfgov_departments tests on each build but got sidetracked with trying to integrate the site with Mandrill and SAML.  If you've got the bandwidth to take that on, please do.

jacine commented on web/modules/custom/sfgov_admin/src/Plugin/Field/FieldWidget/SfgovParagraphsWidget.php:
```diff
@@ -0,0 +1,101 @@
+          '#attached' => ['library' => ['sfgov_admin/paragraphs']],
+          ],
+        ]] + $elements["operations"]["#links"];
+      $elements['#suffix'] = $this->t('to %type', ['%type' => $this->fieldDefinition->getLabel()]);
```
This suffix is not wanted. Can you please remove?

Jaesin replied:
The suffix has been removed.

jacine commented on web/modules/custom/sfgov_admin/src/Plugin/Field/FieldWidget/SfgovParagraphsWidget.php:
```diff
@@ -0,0 +1,101 @@
+ *
+ * @FieldWidget(
+ *   id = "sfgov_paragraphs",
+ *   label = @Translation("Paragraphs EXPERIMENTAL (Custom)"),
```
Maybe we should use `(SF.gov)` instead of `(custom)` in the label so there is a clear association to the machine name?

Jaesin replied:
I changed it.

Jaesin commented:
I added one unit test but when it began to get difficult to mock up the moving components I moved to writing a behat scenario. I could have written tests in drupals testing frameworks but since they aren't currently running and it take more time to set up their features, Behat was easier.

Jaesin commented:
@josh-chou I added Behat tests and enabled this new widget for transactions pages. (node/add/transaction). I will add QA instructions to [SG-183].
